runtime/ints.c

```diff
@@ -601,9 +599,9 @@ CAMLprim value caml_int64_of_string(value s)
   if (signedness) {
     /* Signed representation expected, allow -2^63 to 2^63 - 1 only */
     if (sign >= 0) {
-      if (res >= (uint64_t)1 << 63) caml_failwith(INT64_ERRMSG);
+      if (res > (uint64_t)INT64_MAX) caml_failwith(INT64_ERRMSG);
     } else {
-      if (res >  (uint64_t)1 << 63) caml_failwith(INT64_ERRMSG);
+      if (res > (uint64_t)INT64_MAX + UINT64_C(1)) caml_failwith(INT64_ERRMSG);
```

Comment:
Are these two changes necessary? They make the code harder to read for me.

Reply:
(Just to clarify: the github UI shows only one changed line above my comment, but there is a similar change right above in the diff.)

Reply:
> Are these two changes necessary?

No.

> They make the code harder to read for me.
I hoped to convey more meaning with a name, and show that there's an error if `res` is outside the range of `int64_t`.

The second one can be written as `(uint64_t)INT64_MAX + 1` if that's nicer.

Reply:
I've reverted this change, and found this one-liner (which I've not included) for the whole block in the meantime:

```c
/* Signed representation expected, allow -2^63 to 2^63 - 1 only */
if (res > (uint64_t)INT64_MAX + (sign < 0))
  caml_failwith(INT64_ERRMSG);
```